synapse/federation/transport/client.py

```diff
@@ -669,7 +669,50 @@ async def claim_client_keys(
         path = _create_v1_path("/user/keys/claim")
 
         return await self.client.post_json(
-            destination=destination, path=path, data=query_content, timeout=timeout
+            destination=destination,
+            path=path,
+            data={"one_time_keys": query_content},
+            timeout=timeout,
+        )
+
+    async def claim_client_keys_unstable(
+        self, destination: str, query_content: JsonDict, timeout: Optional[int]
+    ) -> JsonDict:
+        """Claim one-time keys for a list of devices hosted on a remote server.
+
+        Request:
+            {
+              "one_time_keys": {
+                "<user_id>": {
+                  "<device_id>": {"<algorithm>": <count>}
+                }
+              }
+            }
+
+        Response:
+            {
+              "device_keys": {
```

Comment:
This should be `one_time_keys`, no?

Reply:
Hah, yes. Seems like the old code above also has this wrong. 👍  Good catch.

I think the `key_base64` i also bogus, seems like it should just be the key JSON.